main.py

```diff
@@ -648,6 +653,11 @@ def handle_link_up(self, interface):
                 self.notify_topology_update()
                 self.update_instance_metadata(link)
                 self.notify_link_status_change(link, reason='link up')
+        else:
+            link.update_metadata('last_status_change', time.time())
+            self.notify_topology_update()
+            self.update_instance_metadata(link)
+            self.notify_link_status_change(link, reason='link up')
```

Comment:
This way the notifications for the link will happen twice, once for each interface.

Reply:
yep, I think adding a lock here and perform a similar last_status_change that we've got but with a smaller interval should remedy this, I'll push a commit and test this out.

Reply:
@ajoaoff, I got the chance to further debug this here, it turns out that because of the early returns [here](https://github.com/kytos-ng/topology/blob/fix/issue_34/main.py#L634) and [here](https://github.com/kytos-ng/topology/blob/fix/issue_34/main.py#L640) that we have, and since the current interface of the event will try to be [activated](https://github.com/kytos-ng/topology/blob/fix/issue_34/main.py#L630), only one of the interfaces that belong to a link will execute the else branch, for instance, check this execution log (I added an extra line to print out the Link), using this topology:

```
mininet> net
s1 lo:  s1-eth1:h1-eth0 s1-eth2:s2-eth2
s2 lo:  s2-eth1:h2-eth0 s2-eth2:s1-eth2 s2-eth3:s3-eth2
s3 lo:  s3-eth1:h3-eth0 s3-eth2:s2-eth3

```

```
2022-01-06 16:55:54,476 - INFO [kytos.napps.kytos/topology] (ThreadPoolExecutor-0_164) placeholder else branch for link Link(Interface('s1-eth2', 2, Switch('00:00:00:00:00:00:00:01')),
Interface('s2-eth2', 2, Switch('00:00:00:00:00:00:00:02')))

2022-01-06 16:55:54,501 - INFO [kytos.napps.kytos/topology] (ThreadPoolExecutor-0_45) placeholder else branch for link Link(Interface('s3-eth2', 2, Switch('00:00:00:00:00:00:00:03')), I
nterface('s2-eth3', 3, Switch('00:00:00:00:00:00:00:02')))
```

Let me know if you've spotted any other cases.

Reply:
My point is that if the `else` is executed, then the link was already active, so there is no change in its status, no need to notify.

Reply:
@ajoaoff we still need this notification since it's the last opportunity to notify the subscribers since we've optimized not to constantly keep broadcasting [`is.nni` on `add_links`, we have an `if created` conditional there](https://github.com/kytos-ng/topology/blob/master/main.py#L691-L714), so without the notification it would lead to issues on `pathfinder`.

Reply:
@ajoaoff , we'll go ahead and merge this since we've chatted today and the other point is ready to review on PR #48, really appreciated your review and spotting also this other event